Let me know if to working my friend

KennyCoders opened the pull request:
Is to work?

KennyCoders committed:
Update main.py

KennyCoders committed:
Dolev's Card

KennyCoders committed:
Update main.py

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+class Car:
+    def __init__(self, WheelNumber, PSI, FuelType ):
```
not according to python code standards:
1.  all names of variables should start with camelCase letters.
2. when variable name contains more that one word - words should be separated with underline ( _ )

KennyCoders replied:
I have done thy bidding master

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+class Car:
+    def __init__(self, WheelNumber, PSI, FuelType ):
+        self.WheelNumber = WheelNumber
```
very bad name choice! 
the variable must ALWAYS describe correctly the object.
it implies that this variable describe the serial ID of a wheel, but this is not what you meant.
In addition, it's a common beginner's  mistake to try to use short names as possible.
This approach is wrong, because readability is one of the most important factors of your code writing.
you **write** a line of code one time, but the number of times you will **read** it in the future (or other people), will be always greater - so make sure your code is easy to read and understand.

In that case, for example 'number_of_wheels' is a much better choice.

KennyCoders replied:
Done

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+class Car:
+    def __init__(self, WheelNumber, PSI, FuelType ):
+        self.WheelNumber = WheelNumber
+        self.PSI = PSI
```
lowercase letters

KennyCoders replied:
Done

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+    def __init__(self, WheelNumber, PSI, FuelType ):
+        self.WheelNumber = WheelNumber
+        self.PSI = PSI
+        self.FuelType = FuelType
```
lowercase letters + in the future we will limit the options of the fuel types to a certain list (enum), I will write about it later.

KennyCoders replied:
done

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+    def drive(self):
+            print("Wallah im driving")
+
+car = Car("4", "30", "Diesel")
```
why did you pass nubmer parameters as a string?

dolev7 replied:
notice that all of this code section and below is inside of the car class. 
This doesn't make sense, it should be in a different section of course.
In your case, it would be inside def _main_

KennyCoders replied:
Corrected first part,
Did not understand the 2nd point :(

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+class Car:
+    def __init__(self, WheelNumber, PSI, FuelType ):
```
ALWAYS in ANY function you write, you should declare what is the 'type' of each argument. 
https://docs.python.org/3/library/typing.html (read only the first section, see the example in their first use case).

It will not be forced by python that the actual arguments sent to this function would be from this type (because python by design is very not limiting and tend to think the programmer knows what he's doing), but it will be easier for the one who uses your function to understand what is does (in this case the function is the Car constructor and the user is yourself, but you should get the habit as if you are writing your code to other people to read and use)

KennyCoders replied:
Done

dolev7 commented on main.py:
```diff
@@ -1,11 +1,36 @@
+            return self.FuelType
+
+    def show(self):
+        print(self.WheelNumber, self.PSI, self.FuelType)
```
this print doesn't make much sense, I think that you wanted to use something like 'format' and then it will be easier to write things around the arguments.
anyway I didn't define how to print it and you enlarged your head anyway, so no complains really.

Read about this:
https://www.programiz.com/python-programming/string-interpolation

KennyCoders replied:
It's how it was shown in the video :D 
(I also didnt fully understand it)
Monkey see, Monkey do

dolev7 reviewed (changes requested):
Please change this before moving to second part of the OOP assignment :)

KennyCoders committed:
Updated Test - 2

- All variables now start with lower case
- All variables containing more than one word are now separated by '_'

KennyCoders committed:
Update 3

- Updated variable name to 'number_of_wheels'
- Number parameters are no longer strings
- Type of argument is now declared

KennyCoders committed:
Test 3

- Created 3 separate files - Car / Vehicle / Modules.

-Created parent class Vehicle, Child class Car

-Added Max PSI and MAX speed parameters

- Defined a better drive function

- Added Fill up tires function

- Changed Fuel type to Enum (new class)

- Car class now inherits from Vehicle class

- Added number of doors parameter to Car class

* Need to add a more defined Pump Tires function

* Need to complete Modules page to showcase all the different functions

Co-Authored-By: Daniel Dolev <68538339+dolev7@users.noreply.github.com>

KennyCoders committed:
Completed Task pt.2

Please to Checking

Co-Authored-By: Daniel Dolev <68538339+dolev7@users.noreply.github.com>

dolev7 commented on Car.py:
```diff
@@ -0,0 +1,11 @@
+
+
+
+# (self, speed_of_vehicle=, max_speed_of_vehicle, psi, max_psi, number_of_wheels, fuel_type, number_of_doors):
```
no need to leave lines that you commented out

KennyCoders replied:
Comments removed

dolev7 commented on Car.py:
```diff
@@ -0,0 +1,11 @@
+from Vehicle import Vehicle
+
+class Car(Vehicle):
+    def __init__(self, speed_of_vehicle=100, max_speed_of_vehicle=150, psi=30, max_psi=50, number_of_wheels=4, fuel_type=Vehicle.FuelType.GASOLINE, number_of_doors=5):
```
I thought we agreed on leaving the speed out :)

KennyCoders replied:
Agreed.

dolev7 commented on Vehicle.py:
```diff
@@ -0,0 +1,56 @@
+        self.fuel_type = fuel_type
+
+
+# Drive Function:
```
you don't need to explain in the comment the most self-explaining function in world

KennyCoders replied:
Comments removed

dolev7 commented on Vehicle.py:
```diff
@@ -0,0 +1,56 @@
+        print("Walla Vehicle is driving at " + str(speed_of_vehicle) + " KMH")
+
+
+    def increase_speed_of_vehicle(self, amount_of_speed_increase: int):
```
bad function name. 
The function is within the class 'vehicle', so obviously if you increase the speed, it's the speed of the vehicle, no need to repeat twice. 
It makes it look funny when you use it on the vehicle object:
"vehicle1.increase_speed_of_vehicle"

KennyCoders replied:
Name updated

dolev7 commented on Vehicle.py:
```diff
@@ -0,0 +1,56 @@
+    def drive(self, speed_of_vehicle: int):
+        if speed_of_vehicle > self.max_speed_of_vehicle:
+            raise ValueError("Current speed is above the maximum possible speed")
+        self.speed_of_vehicle = speed_of_vehicle
```
according to agreed changes, no need to keep it as class attribute

KennyCoders replied:
Noted.

dolev7 commented on Vehicle.py:
```diff
@@ -0,0 +1,56 @@
+        print(f"Vehicle is now driving in  {self.speed_of_vehicle}  KMH")
+
+
+    def decrease_speed_of_vehicle(self, amount_of_speed_decrease: int):
```
I liked that you used 2 functions to change the speed of vehicle, it's a good practice. 
BUT, to make 2 functions to "decrease" and "increase" speed is a bad idea.
you could just combine them into one function ("update_speed" / "change_speed").

dolev7 replied:
I want to mention that I liked 'amount_of_speed_decrease', great name :)

dolev7 commented on Modules.py:
```diff
@@ -0,0 +1,17 @@
+from Vehicle import Vehicle
```
Actually Modles.py is not the right name for this files.
"Modules" describe additional code that you use, so in this case your modules are Car and Vehicle.
This file doesn't **contain** modules, but only **uses** them.
you can change it's name to main.py or whatever you want, but it's definitely not a Modules file.

KennyCoders replied:
File name Updated

dolev7 commented on Modules.py:
```diff
@@ -0,0 +1,17 @@
+vehicle1.decrease_speed_of_vehicle(40)
+
+
+car3 = Car(100,200,40,60,4,Vehicle.FuelType.PATROL,4)
```
It's ok if you don't change it now, but you should know for the future that in order to improve readability of your code, you can mention the arguments by name, then it will be easier for you when you read it again to know what goes where.
In example: 
car3 = Car(max_speed_of_vehicle = 200, psi = 40, max_psi = 60...)

KennyCoders replied:
Noted for further reference.

dolev7 commented on Modules.py:
```diff
@@ -0,0 +1,17 @@
+
+car3 = Car(100,200,40,60,4,Vehicle.FuelType.PATROL,4)
+car3.increase_speed_of_vehicle(50)
+car3.pump_tires(20)
```
I loved the use of the 'raise', which allows you to "throw" errors, the problem is you don't catch them anywhere.
I started discussing this mechanism with you, you can read about it here:
https://www.w3schools.com/python/python_try_except.asp
ask me if something is not clear :)

KennyCoders replied:
Updated Try/Expect block. ( Only did so for Drive function for training purposes)

dolev7 commented on Vehicle.py:
```diff
@@ -0,0 +1,56 @@
+            raise ValueError("You cannot drive below 0 KMH")
+
+
+#Pump Tires Function:
```
here as well. don't need to explain in comment what the function does if "your explanation" == "the name of the function"

KennyCoders replied:
Comment removed

dolev7 commented on Vehicle.py:
```diff
@@ -0,0 +1,56 @@
+    def decrease_speed_of_vehicle(self, amount_of_speed_decrease: int):
+        self.amount_of_speed_decrease = amount_of_speed_decrease
+        self.speed_of_vehicle -= amount_of_speed_decrease
+        print(f"Vehicle is now driving in  {self.speed_of_vehicle} KMH")
```
2 spaces in print before {self.speed_of_vehicle}

KennyCoders replied:
Fixed

dolev7 reviewed (changes requested):
Good job!

KennyCoders committed:
Dolev you can Ignore this one

User Input Update

KennyCoders committed:
Updated V.4

-Removed unnecessary comments and notes
-Changed  increase_speed_of_vehicle function name
-Changed modules file name
-Improved the Try/Except block
